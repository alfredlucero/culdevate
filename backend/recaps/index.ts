import { NowRequest, NowResponse } from "@now/node";

const sampleRecaps = [
  {
    title: "Summaries and Goals",
    subtitle: "Subtitle: Position, Emphasis, Etc.",
    bulletPoints: ["I endeavor to achieve something!", "Software architect!"],
    type: "objectives",
    startDate: "2019/04/20",
    endDate: null,
  },
  {
    title: "Sandia National Labs",
    subtitle: "Subtitle: Position, Emphasis, Etc.",
    bulletPoints: ["Web Developer R&D Intern", "Software Development Technical Intern"],
    type: "work",
    startDate: "2016/06/20",
    endDate: "2017/06/10",
  },
  {
    title: "University of California, Los Angeles",
    subtitle: "BS in Computer Science, 3.56 GPA",
    bulletPoints: ["Upsilon Pi Epsilon Honor Society 2016-2017", "Daily Bruin Web Developer Intern"],
    type: "education",
    startDate: "2013/10/01",
    endDate: "2017/6/10",
  },
  {
    title: "SJB Tennis",
    subtitle: "Spring Season",
    bulletPoints: ["Varsity Tennis all 4 years", "Captain - Senior Year", "MVP - Third Year"],
    type: "extracurricular",
    startDate: "2009/09/15",
    endDate: "2013/5/20",
  },
  {
    title: "Terraform",
    subtitle: "Subtitle: Position, Emphasis, Etc.",
    bulletPoints: ["Learned how to do infrastructure as code with AWS stuff!"],
    type: "skills",
    startDate: "2018/01/10",
    endDate: null,
  },
  {
    title: "Promoted to Software Engineer II",
    subtitle: "Subtitle: Position, Emphasis, Etc.",
    bulletPoints: ["After 1 year and many projects, I earned my stripes!"],
    type: "achievements",
    startDate: "2018/10/20",
    endDate: null,
  },
  {
    title: "Arm in Arm Award",
    subtitle: "Subtitle: Position, Emphasis, Etc.",
    bulletPoints: ["Helped out CGE with Billing Rewrite to steer them back on time."],
    type: "awards",
    startDate: "2019/01/07",
    endDate: "2019/05/20",
  },
  {
    title: "Reference from Sandia Mentors",
    subtitle: "Subtitle: Position, Emphasis, Etc.",
    bulletPoints: ["Can check them out on LinkedIn"],
    type: "proof",
    startDate: "2016/06/20",
    endDate: "2017/06/05",
  },
  {
    title: "Code Complete Book",
    subtitle: "Subtitle: Position, Emphasis, Etc.",
    bulletPoints: ["Finished reading this book!"],
    type: "other",
    startDate: "2018/10/12",
    endDate: "2019/07/25",
  },
];

module.exports = async (req: NowRequest, res: NowResponse) => {
  console.log(req);

  const test = await Promise.resolve({ blah: "test" });

  const blah = { test: "mate",

};
console.log(blah);

  console.log(test);
  res.json(sampleRecaps);
};
