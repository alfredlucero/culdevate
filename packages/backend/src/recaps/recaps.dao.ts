import {
  RecapBaseModel,
  RecapWorkExperienceModel,
  RecapEducationModel,
  RecapAccomplishmentsModel,
  RecapPublicationsModel,
  RecapSkillsModel,
  RecapSideProjectsModel,
  RecapOrganizationsModel,
  RecapReferencesModel,
  RecapOtherModel,
  Recap,
} from "./recaps.model";

const RecapsDao = {
  findAllRecaps(userId: string) {
    return RecapBaseModel.find({ userId });
  },

  findRecapById(recapId: string) {
    return RecapBaseModel.findById(recapId);
  },

  findRecapByIdAndUserId({ recapId, userId }: { recapId: string; userId: string }) {
    return RecapBaseModel.findOne({ userId, _id: recapId });
  },

  createRecap(recap: Recap) {
    switch (recap.kind) {
      case "Work Experience":
        const workExperienceRecap = new RecapWorkExperienceModel(recap);
        return workExperienceRecap.save();
      case "Education":
        const educationRecap = new RecapEducationModel(recap);
        return educationRecap.save();
      case "Accomplishments":
        const accomplishmentsRecap = new RecapAccomplishmentsModel(recap);
        return accomplishmentsRecap.save();
      case "Publications":
        const publicationsRecap = new RecapPublicationsModel(recap);
        return publicationsRecap.save();
      case "Skills":
        const skillsRecap = new RecapSkillsModel(recap);
        return skillsRecap.save();
      case "Side Projects":
        const sideProjectsRecap = new RecapSideProjectsModel(recap);
        return sideProjectsRecap.save();
      case "Organizations":
        const organizationsRecap = new RecapOrganizationsModel(recap);
        return organizationsRecap.save();
      case "References":
        const referencesRecap = new RecapReferencesModel(recap);
        return referencesRecap.save();
      case "Other":
        const otherRecap = new RecapOtherModel(recap);
        return otherRecap.save();
      default:
        throw new Error("Failed to find matching recap kind to create!");
    }
  },

  updateRecapById({ recapId, updatedRecap }: { recapId: string; updatedRecap: Recap }) {
    switch (updatedRecap.kind) {
      case "Work Experience":
        return RecapWorkExperienceModel.findOneAndUpdate(
          { _id: recapId },
          { $set: updatedRecap, ...(!updatedRecap.endDate ? { $unset: { endDate: 1 } } : {}) },
          { new: true },
        );
      case "Education":
        return RecapEducationModel.findOneAndUpdate(
          { _id: recapId },
          { $set: updatedRecap, ...(!updatedRecap.endDate ? { $unset: { endDate: 1 } } : {}) },
          { new: true },
        );
      case "Accomplishments":
        return RecapAccomplishmentsModel.findOneAndUpdate({ _id: recapId }, updatedRecap, { new: true });
      case "Publications":
        return RecapPublicationsModel.findOneAndUpdate({ _id: recapId }, updatedRecap, { new: true });
      case "Skills":
        return RecapSkillsModel.findOneAndUpdate({ _id: recapId }, updatedRecap, { new: true });
      case "Side Projects":
        return RecapSideProjectsModel.findOneAndUpdate(
          { _id: recapId },
          { $set: updatedRecap, ...(!updatedRecap.endDate ? { $unset: { endDate: 1 } } : {}) },
          { new: true },
        );
      case "Organizations":
        return RecapOrganizationsModel.findOneAndUpdate(
          { _id: recapId },
          { $set: updatedRecap, ...(!updatedRecap.endDate ? { $unset: { endDate: 1 } } : {}) },
          { new: true },
        );
      case "References":
        return RecapReferencesModel.findOneAndUpdate({ _id: recapId }, updatedRecap, { new: true });
      case "Other":
        return RecapOtherModel.findOneAndUpdate({ _id: recapId }, updatedRecap, { new: true });
      default:
        throw new Error("Failed to find matching recap kind to update!");
    }
  },

  removeRecapById(recapId: string) {
    return RecapBaseModel.findByIdAndDelete({ _id: recapId });
  },
};

export default RecapsDao;
