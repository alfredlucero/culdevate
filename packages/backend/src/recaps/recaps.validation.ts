import Joi from "@hapi/joi";

// TODO: replace with yup for validation

export const MAX_BULLETPOINT_LENGTH = 1000;
export const MAX_BULLETPOINTS = 20;
export const MAX_GENERAL_LENGTH = 254;
export const MAX_URL_LENGTH = 2048;

const RecapWorkExperienceSchema = Joi.object({
  title: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
  location: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
  company: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
  employmentType: Joi.string()
    .valid(
      "Part-Time",
      "Full-Time",
      "Self-Employed",
      "Freelance",
      "Contract",
      "Internship",
      "Apprenticeship",
      "Volunteer",
    )
    .required(),
});

const RecapEducationSchema = Joi.object({
  school: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
  location: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
  degree: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
  fieldOfStudy: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
  grade: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
});

const RecapAccomplishmentsSchema = Joi.object({
  title: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
  type: Joi.string()
    .valid("Personal", "Service", "Featured", "School", "Career", "Other")
    .required(),
});

const RecapPublicationsSchema = Joi.object({
  title: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
  type: Joi.string()
    .valid("Book", "Journal", "Newspaper", "Magazine", "Blog", "Other")
    .required(),
  coauthors: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
  publisher: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
  url: Joi.string()
    .allow("")
    .max(MAX_URL_LENGTH),
});

const RecapSkillsSchema = Joi.object({
  title: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
  proficiency: Joi.string()
    .valid("Novice", "Intermediate", "Advanced", "Expert")
    .required(),
});

const RecapSideProjectsSchema = Joi.object({
  title: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
  creators: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
});

const RecapOrganizationsSchema = Joi.object({
  organizationName: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
  location: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
  positions: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
});

const RecapReferencesSchema = Joi.object({
  company: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
  title: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
  phoneNumber: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
  email: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
});

const RecapOtherSchema = Joi.object({
  title: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
});

export const RecapSchema = Joi.object({
  startDate: Joi.date().optional(),
  endDate: Joi.date().optional(),
  bulletPoints: Joi.array()
    .items(Joi.string().max(MAX_BULLETPOINT_LENGTH))
    .max(MAX_BULLETPOINTS)
    .required(),
  kind: Joi.string()
    .valid(
      "Work Experience",
      "Education",
      "Accomplishments",
      "Publications",
      "Skills",
      "Side Projects",
      "Organizations",
      "References",
      "Other",
    )
    .required(),
})
  .when(Joi.object({ kind: "Work Experience" }).unknown(), {
    then: RecapWorkExperienceSchema,
  })
  .when(Joi.object({ kind: "Education" }).unknown(), {
    then: RecapEducationSchema,
  })
  .when(Joi.object({ kind: "Accomplishments" }).unknown(), {
    then: RecapAccomplishmentsSchema,
  })
  .when(Joi.object({ kind: "Publications" }).unknown(), {
    then: RecapPublicationsSchema,
  })
  .when(Joi.object({ kind: "Skills" }).unknown(), {
    then: RecapSkillsSchema,
  })
  .when(Joi.object({ kind: "Side Projects" }).unknown(), {
    then: RecapSideProjectsSchema,
  })
  .when(Joi.object({ kind: "Organizations" }).unknown(), {
    then: RecapOrganizationsSchema,
  })
  .when(Joi.object({ kind: "References" }).unknown(), {
    then: RecapReferencesSchema,
  })
  .when(Joi.object({ kind: "Other" }).unknown(), {
    then: RecapOtherSchema,
  });
