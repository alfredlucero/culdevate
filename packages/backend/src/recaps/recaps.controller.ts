import { Response } from "express";
import { RequestWithUser } from "../interfaces/requestWithUser";
import RecapsDao from "./recaps.dao";
import { Recap } from "./recaps.model";

const RecapsController = {
  async createRecap(req: RequestWithUser, res: Response) {
    const currentUser = req.user;
    const recap: Recap = {
      ...req.body,
      userId: currentUser.id,
    };

    try {
      const createdRecap = await RecapsDao.createRecap(recap);

      res.status(201).json(createdRecap);
    } catch (err) {
      res.status(500).json({ message: "Failed to create recap." });
    }
  },

  async updateRecap(req: RequestWithUser, res: Response) {
    const currentUser = req.user;
    const recapId = req.params.recapId;
    const updatedRecap: Recap = {
      ...req.body,
      userId: currentUser.id,
    };

    try {
      const foundRecap = await RecapsDao.findRecapById(recapId);
      const isMatchingUserId = foundRecap && foundRecap.userId.toString() === currentUser.id;

      if (!foundRecap || !isMatchingUserId) {
        return res.status(404).json({ message: "Failed to find matching recap to update" });
      }

      const changedRecap = await RecapsDao.updateRecapById({ recapId, updatedRecap });

      if (changedRecap) {
        res.status(200).json(changedRecap);
      } else {
        // If by some chance it got deleted while we were trying to update it
        // we'll still get a 404 back
        res.status(404).json({ message: "Failed to find matching recap to update" });
      }
    } catch (err) {
      res.status(500).json({ message: "Failed to update recap" });
    }
  },

  async deleteRecap(req: RequestWithUser, res: Response) {
    const recapId = req.params.recapId;
    const currentUser = req.user;

    try {
      const foundRecap = await RecapsDao.findRecapById(recapId);
      const isMatchingUserId = foundRecap && foundRecap.userId.toString() === currentUser.id;

      if (!foundRecap || !isMatchingUserId) {
        return res.status(404).json({ message: "Failed to find matching recap to delete" });
      }

      const deletedRecap = await RecapsDao.removeRecapById(recapId);

      if (deletedRecap) {
        res.status(200).json(deletedRecap);
      } else {
        // If by some chance it got deleted while we were trying to delete it
        // we'll still get a 404 back
        res.status(404).json({ message: "Failed to find matching recap to delete" });
      }
    } catch (err) {
      res.status(500).json({ message: "Failed to delete recap" });
    }
  },
};

export default RecapsController;
