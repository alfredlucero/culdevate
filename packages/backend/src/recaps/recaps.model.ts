import mongoose from "mongoose";
import { MAX_BULLETPOINT_LENGTH, MAX_BULLETPOINTS, MAX_GENERAL_LENGTH } from "./recaps.validation";

export interface RecapBase {
  kind: RecapKind;
  startDate?: Date;
  endDate?: Date;
  bulletPoints: string[];
}

export type RecapKind =
  | "WorkExperience"
  | "Education"
  | "Accomplishments"
  | "Skills"
  | "SideProjects"
  | "Organizations"
  | "References"
  | "Other";

const recapBaseOptions = {
  discriminatorKey: "kind",
  collection: "recaps",
};

const recapBaseSchema = new mongoose.Schema(
  {
    startDate: { type: Date },
    endDate: { type: Date },
    bulletPoints: {
      type: [
        {
          type: String,
          maxlength: MAX_BULLETPOINT_LENGTH,
        },
      ],
      required: true,
      maxlength: MAX_BULLETPOINTS,
    },
  },
  recapBaseOptions,
);

export interface RecapBaseModel extends RecapBase, mongoose.Document {}

// Base Recap model from which other Recap models will inherit from based on the
// the value of the "kind" discriminator
export const RecapBaseModel = mongoose.model<RecapBaseModel>("Recap", recapBaseSchema);

// Work Experience

export interface RecapWorkExperience {
  title: string;
  location: string;
  company: string;
  employmentType: EmploymentType;
}

export type EmploymentType = "partTime" | "fullTime";

const recapWorkExperienceSchema = new mongoose.Schema({
  title: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
  location: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
  company: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
  employmentType: { type: String, required: true, enum: ["partTime", "fullTime"] },
});

export interface RecapWorkExperienceModel extends RecapBaseModel, RecapWorkExperience {}

const RecapWorkExperience = RecapBaseModel.discriminator("WorkExperience", recapWorkExperienceSchema);

export const RecapWorkExperienceModel = mongoose.model<RecapWorkExperienceModel>("WorkExperience");

// Education

export interface RecapEducation {
  school: string;
  location: string;
  degree: string;
  fieldOfStudy: string;
  grade: string;
}

const recapEducationSchema = new mongoose.Schema({
  school: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
  location: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
  degree: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
  fieldOfStudy: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
  grade: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
});

export interface RecapEducationModel extends RecapBaseModel, RecapEducation {}

const RecapEducation = RecapBaseModel.discriminator("Education", recapEducationSchema);

export const RecapEducationModel = mongoose.model<RecapEducationModel>("Education");

// Accomplishments

export interface RecapAccomplishments {
  title: string;
  type: string;
}

const recapAccomplishmentsSchema = new mongoose.Schema({
  title: {
    type: String,
    required: true,
    maxlength: MAX_GENERAL_LENGTH,
  },
  type: {
    type: String,
    required: true,
    maxlength: MAX_GENERAL_LENGTH,
  },
});

export interface RecapAccomplishmentsModel extends RecapBaseModel, RecapAccomplishments {}

const RecapAccomplishments = RecapBaseModel.discriminator("Accomplishments", recapAccomplishmentsSchema);

export const RecapAccomplishmentsModel = mongoose.model<RecapAccomplishmentsModel>("Accomplishments");

// Skills

export interface RecapSkills {
  title: string;
}

const recapSkillsSchema = new mongoose.Schema({
  title: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
});

export interface RecapSkillsModel extends RecapBaseModel, RecapSkills {}

const RecapSkills = RecapBaseModel.discriminator("Skills", recapSkillsSchema);

export const RecapSkillsModel = mongoose.model<RecapSkillsModel>("Skills");

// Side Projects

export interface RecapSideProjects {
  title: string;
}

const recapSideProjectsSchema = new mongoose.Schema({
  title: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
});

export interface RecapSideProjectsModel extends RecapBaseModel, RecapSideProjects {}

const RecapSideProjects = RecapBaseModel.discriminator("SideProjects", recapSideProjectsSchema);

export const RecapSideProjectsModel = mongoose.model<RecapSideProjectsModel>("SideProjects");

// Organizations

export interface RecapOrganizations {
  organizationName: string;
  location: string;
  positions: string;
}

const recapOrganizationsSchema = new mongoose.Schema({
  organizationName: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
  location: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
  positions: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
});

export interface RecapOrganizationsModel extends RecapBaseModel, RecapOrganizations {}

const RecapOrganizations = RecapBaseModel.discriminator("Organizations", recapOrganizationsSchema);

export const RecapOrganizationsModel = mongoose.model<RecapOrganizationsModel>("Organizations");

// References

export interface RecapReferences {
  company: string;
  title: string;
  phoneNumber: string;
  email: string;
}

export const recapReferencesSchema = new mongoose.Schema({
  company: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
  title: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
  phoneNumber: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
  email: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
});

export interface RecapReferencesModel extends RecapBaseModel, RecapReferences {}

const RecapReferences = RecapBaseModel.discriminator("References", recapReferencesSchema);

export const RecapReferencesModel = mongoose.model<RecapReferencesModel>("References");

// Other

export interface RecapOther {
  title: string;
}

const recapOtherSchema = new mongoose.Schema({
  title: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
});

export interface RecapOtherModel extends RecapBaseModel, RecapOther {}

const RecapOther = RecapBaseModel.discriminator("Other", recapOtherSchema);

export const RecapOtherModel = mongoose.model<RecapOtherModel>("Other");
