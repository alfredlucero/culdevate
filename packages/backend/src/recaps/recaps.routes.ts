import { Router } from "express";
import RecapsController from "./recaps.controller";
import authMiddleware from "../middleware/auth.middleware";
import validationMiddleware from "../middleware/validation.middleware";
import { RecapSchema } from "./recaps.validation";

const RecapsRoutes = {
  path: "/recaps",
  router: Router(),

  initializeRoutes() {
    // POST /recaps
    this.router.post("/", authMiddleware, validationMiddleware(RecapSchema, "body"), RecapsController.createRecap);

    // PATCH /recaps/:recapId
    this.router.patch(
      "/:recapId",
      authMiddleware,
      validationMiddleware(RecapSchema, "body"),
      RecapsController.updateRecap,
    );

    return this.router;
  },
};

export default RecapsRoutes;
