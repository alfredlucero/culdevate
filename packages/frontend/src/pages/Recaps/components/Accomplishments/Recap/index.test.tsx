import React from "react";
import { render } from "@testing-library/react";
import { RecapAccomplishments } from "../../../../../interfaces/recaps.interface";
import AccomplishmentsRecap from "./index";

const accomplishments: RecapAccomplishments = {
  title: "Promoted to Software Engineer 2 at SendGrid",
  kind: "Accomplishments",
  userId: "userId",
  _id: "accomplishmentsId",
  type: "Career",
  bulletPoints: [
    "Led migration of manual frontend deployments and hosting on on-prem nginx servers to AWS S3 and CloudFront with Terraform and Buildkite for CICD",
    "Contributed to the development and pushed the final release of the redesigned Email Activity in Backbone/Marionette",
    "Led the transition from an in-house Ruby Selenium solution to WebdriverIO and finally to Cypress for E2E tests",
  ],
  startDate: new Date("2018/10/20"),
};

describe("<AccomplishmentsRecap />", () => {
  const testId = "accomplishmentsRecapTestId";

  test("should render without error", () => {
    const { container } = render(
      <AccomplishmentsRecap
        accomplishments={accomplishments}
        onEdit={() => {}}
        onDelete={() => {}}
        testId={testId}
        className="extra-recap-class"
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
