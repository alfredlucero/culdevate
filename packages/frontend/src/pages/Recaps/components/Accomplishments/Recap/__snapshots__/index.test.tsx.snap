// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccomplishmentsRecap /> should render without error 1`] = `
<div>
  <div
    class="bg-white shadow-md rounded px-8 py-6 p-4 flex extra-recap-class"
    data-testid="accomplishmentsRecapTestId"
  >
    <div
      class="flex justify-center items-center p-2 rounded-lg bg-gray-200 recap-icon mr-4"
    >
      <div
        class="font-sans text-3xl text-teal-400"
        id=""
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-award fa-w-12 "
          data-icon="award"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M97.12 362.63c-8.69-8.69-4.16-6.24-25.12-11.85-9.51-2.55-17.87-7.45-25.43-13.32L1.2 448.7c-4.39 10.77 3.81 22.47 15.43 22.03l52.69-2.01L105.56 507c8 8.44 22.04 5.81 26.43-4.96l52.05-127.62c-10.84 6.04-22.87 9.58-35.31 9.58-19.5 0-37.82-7.59-51.61-21.37zM382.8 448.7l-45.37-111.24c-7.56 5.88-15.92 10.77-25.43 13.32-21.07 5.64-16.45 3.18-25.12 11.85-13.79 13.78-32.12 21.37-51.62 21.37-12.44 0-24.47-3.55-35.31-9.58L252 502.04c4.39 10.77 18.44 13.4 26.43 4.96l36.25-38.28 52.69 2.01c11.62.44 19.82-11.27 15.43-22.03zM263 340c15.28-15.55 17.03-14.21 38.79-20.14 13.89-3.79 24.75-14.84 28.47-28.98 7.48-28.4 5.54-24.97 25.95-45.75 10.17-10.35 14.14-25.44 10.42-39.58-7.47-28.38-7.48-24.42 0-52.83 3.72-14.14-.25-29.23-10.42-39.58-20.41-20.78-18.47-17.36-25.95-45.75-3.72-14.14-14.58-25.19-28.47-28.98-27.88-7.61-24.52-5.62-44.95-26.41-10.17-10.35-25-14.4-38.89-10.61-27.87 7.6-23.98 7.61-51.9 0-13.89-3.79-28.72.25-38.89 10.61-20.41 20.78-17.05 18.8-44.94 26.41-13.89 3.79-24.75 14.84-28.47 28.98-7.47 28.39-5.54 24.97-25.95 45.75-10.17 10.35-14.15 25.44-10.42 39.58 7.47 28.36 7.48 24.4 0 52.82-3.72 14.14.25 29.23 10.42 39.59 20.41 20.78 18.47 17.35 25.95 45.75 3.72 14.14 14.58 25.19 28.47 28.98C104.6 325.96 106.27 325 121 340c13.23 13.47 33.84 15.88 49.74 5.82a39.676 39.676 0 0 1 42.53 0c15.89 10.06 36.5 7.65 49.73-5.82zM97.66 175.96c0-53.03 42.24-96.02 94.34-96.02s94.34 42.99 94.34 96.02-42.24 96.02-94.34 96.02-94.34-42.99-94.34-96.02z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div
      class="flex-auto"
    >
      <h4
        class="font-mono font-bold text-gray-800 text-xl "
      >
        Promoted to Software Engineer 2 at SendGrid
      </h4>
      <p
        class="font-sans text-gray-800 text-base italic text-gray-600"
      >
        October 20, 2018
      </p>
      <p
        class="font-sans text-gray-800 text-base "
      >
        Career
      </p>
      <ul
        class="mt-3"
      >
        <li
          class="flex items-center"
        >
          <div
            class="font-sans text-2xl mr-2 text-teal-300"
            id=""
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-square fa-w-14 "
              data-icon="square"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
                fill="currentColor"
              />
            </svg>
          </div>
          <p
            class="font-sans text-gray-800 text-base pt-1"
          >
            Led migration of manual frontend deployments and hosting on on-prem nginx servers to AWS S3 and CloudFront with Terraform and Buildkite for CICD
          </p>
        </li>
        <li
          class="flex items-center"
        >
          <div
            class="font-sans text-2xl mr-2 text-teal-300"
            id=""
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-square fa-w-14 "
              data-icon="square"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
                fill="currentColor"
              />
            </svg>
          </div>
          <p
            class="font-sans text-gray-800 text-base pt-1"
          >
            Contributed to the development and pushed the final release of the redesigned Email Activity in Backbone/Marionette
          </p>
        </li>
        <li
          class="flex items-center"
        >
          <div
            class="font-sans text-2xl mr-2 text-teal-300"
            id=""
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-square fa-w-14 "
              data-icon="square"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
                fill="currentColor"
              />
            </svg>
          </div>
          <p
            class="font-sans text-gray-800 text-base pt-1"
          >
            Led the transition from an in-house Ruby Selenium solution to WebdriverIO and finally to Cypress for E2E tests
          </p>
        </li>
      </ul>
    </div>
    <div
      class="w-1/3 text-right"
    >
      <button
        class="font-bold py-2 px-4 rounded bg-transparent text-teal-700 border border-teal-500 hover:bg-teal-500 hover:text-white hover:border-transparent mr-2"
        id="accomplishmentsId"
        type="button"
      >
        <div
          class="flex items-center"
        >
          <div
            class="font-sans text-2xl mr-2"
            id=""
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-pencil-alt fa-w-16 "
              data-icon="pencil-alt"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"
                fill="currentColor"
              />
            </svg>
          </div>
           Edit
        </div>
      </button>
      <button
        class="font-bold py-2 px-4 rounded font-bold py-2 px-4 rounded bg-red-500 text-white hover:bg-red-700 "
        id="accomplishmentsId"
        type="button"
      >
        <div
          class="flex items-center"
        >
          <div
            class="font-sans text-2xl mr-2"
            id=""
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-trash-alt fa-w-14 "
              data-icon="trash-alt"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
                fill="currentColor"
              />
            </svg>
          </div>
          Delete
        </div>
      </button>
    </div>
  </div>
</div>
`;
